olive/evaluator/metric.py

```diff
@@ -47,11 +50,12 @@ class LatencySubType(str, Enum):
 class Metric(ConfigBase):
     name: str
     type: MetricType
-    sub_type: Union[AccuracySubType, LatencySubType] = None
+    sub_type: Union[List[Union[str, AccuracySubType, LatencySubType]], AccuracySubType, LatencySubType] = None
+    sub_type_for_rank: str = None
```

Comment:
I think accepting a list is a good idea since we don't want to waste resources running a new metric for each subtype. 

However, I am not fully sure about the `sub_type_for_rank` idea. This looks like a arbitrary design choice. 

Maybe it is simpler, if we also ask for list for `name`, `priority_rank`, and `goal`? 
How is `goal` resolved and applied with `sub_type_for_rank`?